src/hotspot/share/runtime/thread.cpp

```diff
@@ -482,6 +482,27 @@ void Thread::check_for_dangling_thread_pointer(Thread *thread) {
 }
 #endif
 
+// Is the target JavaThread protected by this Thread:
+bool Thread::is_JavaThread_protected(const JavaThread* p) {
+  if (current() == p) {
+    // Current thread is always protected:
+    return true;
+  }
```

Comment:
This is true but it is awkward that we are going to manifest Thread::current multiple times when using this - see below.